corehq/apps/geospatial/static/geospatial/js/geospatial_map.js

```diff
@@ -282,6 +305,91 @@ hqDefine("geospatial/js/geospatial_map", [
             }
         };
 
+        function resetMarkersOpacity() {
+            let mapInstance = map.getMapboxInstance();
+            let markers = [];
+            Object.keys(caseGroupsIndex).forEach(itemCoordinates => {
+                const mapMarkerItem = caseGroupsIndex[itemCoordinates];
+                markers.push(mapMarkerItem.item);
+
+                const lineId = getLineFeatureId(mapMarkerItem.item.itemId);
+                if (mapInstance.getLayer(lineId)) {
+                    mapInstance.setPaintProperty(lineId, 'line-opacity', 1);
+                }
+            });
+            // It's necessary to delay obscuring the markers since mapbox does not play nice
+            // if we try to do it all at once.
+            setTimeout(function () {
```

Comment:
Maybe create a "change_markers_opacity" function for this? I see a similar pattern on line 352